metals/src/main/scala/scala/meta/internal/metals/codeactions/ImportMissingGiven.scala

```diff
@@ -0,0 +1,86 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals._
+import scala.meta.pc.CancelToken
+
+import org.eclipse.{lsp4j => l}
+
+class ImportMissingGiven(compilers: Compilers) extends CodeAction {
+  override def kind: String = l.CodeActionKind.QuickFix
+
+  override def contribute(
+      params: l.CodeActionParams,
+      token: CancelToken
+  )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]] = {
+
+    def addGivenImport(
+        diagnostic: l.Diagnostic,
+        packageName: String,
+        name: String
+    ): Future[Seq[l.CodeAction]] =
+      importMissingSymbol(diagnostic, packageName, name)
+
+    def importMissingSymbol(
+        diagnostic: l.Diagnostic,
+        packageName: String,
+        name: String
+    ): Future[Seq[l.CodeAction]] = {
+      val textDocumentPositionParams = new l.TextDocumentPositionParams(
+        params.getTextDocument(),
+        diagnostic.getRange.getEnd()
+      )
+      compilers
+        .autoImports(textDocumentPositionParams, name, token)
```

Comment:
`autoImports` aren't yet implemented for Scala 3, so we would have to make do without it.

Reply:
ah, I was just thinking about that. The only difficult thing would be to determine import position